### PR TITLE
Automate manifest update

### PR DESCRIPTION
**Pull request by Mojachieee**
One of the many steps we need to automate our release. 

Create a new action to auto-update manifests when given a version number, it will then PR the changes.

Example PR:
https://github.com/storageos/operator/pull/106

### CODE REVIEW

File: .github/workflows/update-manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+         fi
+         if [[ -n "${{ github.event.client_payload.portal-manager-version }}" ]]; then
+            sed -i 's/PORTAL_MANAGER_VERSION ?= v.*/PORTAL_MANAGER_VERSION ?= ${{ github.event.client_payload.portal-manager-version }}/' Makefile
+            make portal-manager
+         fi
+         if [[ -n "${{ github.event.client_payload.node-manager-version }}" ]]; then
+            sed -i 's/NODE_MANAGER_VERSION ?= v.*/NODE_MANAGER_VERSION ?= ${{ github.event.client_payload.node-manager-version }}/' Makefile
+            make node-manager
+         fi
+      - name: Create Pull Request
```
Comment (mhmxs) [inline .github/workflows/update-manifest.yaml:43]: This nice, and i like it, but i suggest to run this only on PR builds. I don't want to run this flow when we run image build and publish in the same time. So image builds must happen only and if manifests are up to date. I suggest a validation before image build starts.

**Comment by Ricardo-Osorio** [comment]
If there are now automated can we not add those files to .gitignore? We can now forget to update them locally before pushing and it's deal with plus if we care about having those we still have makefile rules to do it...

**Comment by mhmxs** [comment]
> If these are now automated can we add those files to .gitignore? Since we won't have to run it locally and in cases where you need those we still have makefile rules to do it...

There is one disadvantages of generating it, it tooks a long time.

**Comment by Ricardo-Osorio** [comment]
> There is one disadvantages of generating it, it tooks a long time.

They would exist on the repo at all times and always the most up to date they can be.
Although with my limited experience I don't know how much we usually rely on those so it's up to you entirely 👍

File: .github/workflows/update-manifest.yaml (hunk 1 above)
Comment (Mojachieee) [reply .github/workflows/update-manifest.yaml:43]: My initial plan was to make a release of api-manager trigger this pipeline. 

It shouldn't run on push or PR.

@mhmxs how does that sound?

**Review by Ricardo-Osorio — APPROVED** [review]

**Pull request merged**